lib/parse-statements.js

```diff
@@ -85,10 +89,12 @@ function parseImport(result, atRule) {
       if (
         prev.type !== "comment" &&
         (prev.type !== "atrule" ||
-          (prev.name !== "import" && prev.name !== "charset"))
+          (prev.name !== "import" &&
+            prev.name !== "charset" &&
+            !(prev.name === "layer" && !prev.nodes)))
       ) {
         return result.warn(
-          "@import must precede all other statements (besides @charset)",
+          "@import must precede all other statements (besides @charset or empty @layer)",
```

Comment:
Not saying this is wrong, but can you show where the spec allows for empty `@layer` before `@import`?

Reply:
https://drafts.csswg.org/css-cascade-5/#at-import

>Any [@import](https://drafts.csswg.org/css-cascade-5/#at-ruledef-import) rules must precede all other valid at-rules and style rules in a style sheet (ignoring [@charset](https://drafts.csswg.org/css-syntax-3/#at-ruledef-charset) and empty [@layer](https://drafts.csswg.org/css-cascade-5/#at-ruledef-layer) definitions) and must not have any other valid at-rules or style rules between it and previous @import rules, or else the @import rule is invalid.

https://drafts.csswg.org/css-cascade-5/#layer-block

> Note: [@layer](https://drafts.csswg.org/css-cascade-5/#at-ruledef-layer) [block at-rules](https://drafts.csswg.org/css-syntax-3/#block-at-rule) cannot be interleaved with [@import](https://drafts.csswg.org/css-cascade-5/#at-ruledef-import) rules.

In more detail : https://drafts.csswg.org/css-cascade-5/#layer-empty

>The [@layer](https://drafts.csswg.org/css-cascade-5/#at-ruledef-layer) rule can also be used to define new layers without assigning any style rules, by providing only the [layer name](https://drafts.csswg.org/css-cascade-5/#layer-name):
>
>@layer [<layer-name>](https://drafts.csswg.org/css-cascade-5/#typedef-layer-name)[#](https://drafts.csswg.org/css-values-4/#mult-comma);
Such empty [@layer](https://drafts.csswg.org/css-cascade-5/#at-ruledef-layer) rules are allowed before [@import](https://drafts.csswg.org/css-cascade-5/#at-ruledef-import) and @namespace rules (after the [@charset](https://drafts.csswg.org/css-syntax-3/#at-ruledef-charset) rule, if any) as well as everywhere @layer [block at-rules](https://drafts.csswg.org/css-syntax-3/#block-at-rule) are allowed.

------------

But now that I have some distance since writing this I think that `empty @layer` is not a useful way to describe this to stylesheet authors.

_I had to look it up, both to check that I didn't make a mistake and what the exact meaning was._

Is there a better, more widely used name for at-rules without a block?

```css
@layer foo; /* a good name for this? */

@import 'baz.css' layer(baz);
```

Reply:
I don't know if there's a good name for it; I think following the spec's wording is a safe bet.

Reply:
That works for me!
We can also see if this causes confusion to authors and adjust if needed.
It might be a non issue as people learn the feature.

Reply:
All set, just need to merge and release this when I get a chance.